EIPS/eip-rpc.md

```diff
@@ -0,0 +1,2314 @@
+---
+eip: <to be assigned>
+title: Remote procedure call specification
+author: Paul Bouchon <mail@bitpshr.net>
+discussions-to: https://ethereum-magicians.org/t/eip-remote-procedure-call-specification/1537
+status: Draft
+type: Standards Track
+category: Interface
+created: 2018-10-02
+---
+
+## Simple Summary
+
+This proposal defines a standard set of remote procedure call methods that an Ethereum node must implement.
```

Comment:
The `must` in this sentence my need to be changed or wording added to cover RPC implementations which don't implement all methods.

- infura doesn't support stateful methods like `eth_sign` or filters.
- trinity doesn't plan to implement any account support so `eth_sendTransaction` and `eth_sign` won't be supported.

There is discussion to add an "Unsupported" status code which I think is a good way for RPC servers to signal that they do not support an RPC method, however this sentence suggests that not implementing a method would make a JSON-RPC server implementation non-compliant which seems wrong.

- This could be updated to just use `should` instead of `must`
- We could define a subset of the endpoints which are **required** and the remaining ones are *recommended* (aka optional)

Open to other suggestions.

Reply:
Good catch. I changed `must` to `should` and also added an "Unsupported" status code as per @ryanschneider's suggestion above.